patches/tModLoader/Terraria/ModLoader/TMLLootDatabase.cs

```diff
@@ -0,0 +1,831 @@
+using System.Collections.Generic;
+using System.Linq;
+using Terraria.GameContent;
+using Terraria.ID;
+using Terraria.Localization;
+
+namespace Terraria.ModLoader
+{
+	// can and/or will be used for other tml loots later. (ie pots loot)
+	public class TMLLootDatabase
+	{
+		private Dictionary<int, ChestLoot> npcShopByType = new();
+		private List<ChestLoot.Entry> globalNpcShopEntries = new();
+
+		public TMLLootDatabase() {
+			Initialize();
+		}
+
+		public void RegisterNpcShop(int npcId, ChestLoot chestLoot) => npcShopByType.Add(npcId, chestLoot);
+
+		public void RegisterGlobalNpcShop(ChestLoot.Entry entry) => globalNpcShopEntries.Add(entry);
+
+		public ChestLoot GetNpcShopById(int npcId) => npcShopByType[npcId];
+
+		public List<ChestLoot.Entry> GetGlobalNpcShopEntries() => globalNpcShopEntries;
+
+		public void Initialize() {
+			RegisterMerchant();
+			RegisterArmsDealer();
+			RegisterDryad();
+			RegisterBombGuy();
+			RegisterClothier();
+			RegisterGoblin();
+			RegisterWizard();
+			RegisterMechanic();
+			RegisterSantaClaus();
+			RegisterTruffle();
+			RegisterSteampunker();
+			RegisterDyeTrader();
+			RegisterPartyGirl();
+			RegisterCyborg();
+			RegisterPainter();
+			RegisterWitchDoctor();
+			RegisterPirate();
+			RegisterStylist();
+			RegisterSkeletonMerchant();
+			RegisterBartender();
+			RegisterGolfer();
+			RegisterZoologist();
+			RegisterPrincess();
```

Comment:
Double check for updates to shops from 1.4.4
For example Painter has a few extra items I believe